contrib/format-deltalake/src/main/java/org/apache/drill/exec/store/delta/format/DeltaFormatPluginConfig.java

```diff
@@ -18,15 +18,55 @@
 package org.apache.drill.exec.store.delta.format;
 
 import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.annotation.JsonTypeName;
+import org.apache.drill.common.PlanStringBuilder;
 import org.apache.drill.common.logical.FormatPluginConfig;
 
+import java.util.Objects;
+
 @JsonTypeName(DeltaFormatPluginConfig.NAME)
 public class DeltaFormatPluginConfig implements FormatPluginConfig {
 
   public static final String NAME = "delta";
 
+  private final Long version;
+  private final Long timestamp;
+
   @JsonCreator
-  public DeltaFormatPluginConfig() {
+  public DeltaFormatPluginConfig(@JsonProperty("version") Long version,
```

Comment:
I don't remember the annotation, but do you think it would be a good idea to hide these in the config so people don't set them accidentally?

Reply:
I think hiding it could cause issues when serializing / deserializing query plans. But people will not know about this property if it is not specified in the docs for this storage plugin configs.

Reply:
I'm a Jackson contributor - but admit that its not always obvious when and why annotations are needed.
These JsonProperty annotations do seem unnecessary - what happens if you leave them out but keep the rest of the new code?

Reply:
It will fail with the following error:
```
Invalid type definition for type `org.apache.drill.exec.store.delta.format.DeltaFormatPluginConfig`: Argument #0 of constructor [constructor for `org.apache.drill.exec.store.delta.format.DeltaFormatPluginConfig` (2 args), annotations: {interface com.fasterxml.jackson.annotation.JsonCreator=@com.fasterxml.jackson.annotation.JsonCreator(mode=DEFAULT)} has no property name (and is not Injectable): can not use as property-based Creator
```

Reply:
I dislike the need for all the annotations but Jackson insists on when you go down certain paths.

If you put back the empty constructor as well as having the 2 param constructor, you will possibly find that you don't need the JsonCreator and JsonProperty annotations any more.

Jackson seems to prefer to use the no params constructor and to use reflection to set the values on the fields.

Reply:
I propose to leave it as is since it is required to be able to serialize/deserialize these values, and Drill uses the same code for reading plugin configs from the query plan and from UI, where the user sets them.